### DISCUSSION TITLE
How to replicate a complex database with lots of foreign key constraints

### DISCUSSION THREAD

**Discussion by ScottChapman** [osalvador/ReplicaDB#149] [Q&A] [answered]
So I am trying to migrate some old bugzilla data from MySQL -> postgres.
What I find challenging is that the table contraints make it difficult to do this table by table.
For example, when inserting new data I often get an error that the id doesn't exist in the other table (I presume that data is just skipped which isn't terrible). This just makes it difficult because I have to figure out what order to replicate the tables (non-trivial).

The other problem is that I can't truncate the data because of the constraint:
`org.postgresql.util.PSQLException: ERROR: cannot truncate a table referenced in a foreign key constraint`

Is there some reasonable approach I can take to make this easier?
For reference the schema generally looks like this: https://www.bugzilla.org/docs/2.16/html/dbschema.html

**Comment by osalvador** [comment, marked as answer]
To migrate complete databases you need other kind of tool, since ReplicaDB is oriented to replicate data between heterogeneous databases, it is not intended to migrate complete databases. 

I recommend you to check the `pgloeader` tool that is more oriented to this kind of tasks in Postgres. 
https://www.digitalocean.com/community/tutorials/how-to-migrate-mysql-database-to-postgres-using-pgloader

**Reply by ScottChapman** [reply]
Yea that was ultimately what I used, just to get the schema over anyway.
